### PR TITLE
Adds a response controller to OpauthController for the profilecompletion endpoint

### PR DESCRIPTION
**Pull request by nzphoenix**
Updated to use `ModelAsController::controller_for($tmpPage);` as per #38

### CODE REVIEW

File: code/OpauthController.php [hunk 1]
```diff
@@ -52,6 +52,23 @@ public function __construct($dataRecord = null) {
 	}
 
 	/**
+	 * Prepare the controller for handling the response to this request
+	 *
+	 * @param string $title Title to use
+	 * @return Controller
+	 */
+	protected function getResponseController($title) {
```
Comment (dhensby) [inline code/OpauthController.php]: `$title` is not used anywhere that I can see?

File: code/OpauthController.php [hunk 2]
```diff
@@ -52,6 +52,23 @@ public function __construct($dataRecord = null) {
...
+		if(!class_exists('SiteTree')) return $this;
+
+		// Use sitetree pages to render the opauth pages
+		$tmpPage = new Page();
```
Comment (willmorgan) [inline code/OpauthController.php]: I suspect the `$title` param wants to go here

Comment (dhensby) [inline code/OpauthController.php]:
```
array(
    'Title' => $title,
    'ID' => -1,
);
```

Comment (nzphoenix) [inline code/OpauthController.php]: Ah yeah, that sounds about right. I'll take a look, test, and send through another pull request.

File: code/OpauthController.php [hunk 3]
```diff
@@ -52,6 +52,25 @@ public function __construct($dataRecord = null) {
...
+		$tmpPage->ID = -1;
+		$tmpPage->Title = $title;
```
Comment (nzphoenix) [inline code/OpauthController.php]: Makes use of $title.

**Comment by willmorgan** [comment]
Not sure what that merge commit is all about... should it be in this PR?

**Comment by nzphoenix** [comment]
I had started the changes on a different branch. I could see if I can rebase to make it clearer..

**Comment by nzphoenix** [comment]
Ah, doesn't seem I can change it. It was just merging in the previous commit to it. If you'd like it tidier I can close this pull request, merge all the commits into one and re-open as a new pull.

**Comment by dhensby** [comment]
squash and merge is our friend. thanks!

**Pull request merged**